clustermesh-apiserver/ciliumendpointslice/ciliumendpointslice.go

```diff
@@ -0,0 +1,28 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package ciliumendpointslice
+
+import (
+	"github.com/cilium/hive/cell"
+	"github.com/spf13/pflag"
+)
+
+type CiliumEndpointSliceConfig struct {
+	EnableCiliumEndpointSlice bool
+}
+
+func (def CiliumEndpointSliceConfig) Flags(flags *pflag.FlagSet) {
+	flags.Bool("enable-cilium-endpoint-slice", def.EnableCiliumEndpointSlice, "Enables the CiliumEndpointSlice feature")
+}
+
+var DefaultCiliumEndpointSliceConfig = CiliumEndpointSliceConfig{
+	EnableCiliumEndpointSlice: false,
+}
+
+var CiliumEndpointSliceCell = cell.Module(
+	"cilium-endpoint-slice",
+	"Cilium Endpoint Slice",
+
+	cell.Config(DefaultCiliumEndpointSliceConfig),
+)
```

Comment:
Nit. I'd probably directly register the config inside `ResourcesCell`, as somewhat related.